scaleway/resource_domain_record.go

```diff
@@ -0,0 +1,426 @@
+package scaleway
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	domain "github.com/scaleway/scaleway-sdk-go/api/domain/v2beta1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayDomainRecord() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewayDomainRecordCreate,
+		ReadContext:   resourceScalewayDomainRecordRead,
+		UpdateContext: resourceScalewayDomainRecordUpdate,
+		DeleteContext: resourceScalewayDomainRecordDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Default: schema.DefaultTimeout(defaultDomainRecordTimeout),
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"dns_zone": {
+				Type:        schema.TypeString,
+				Description: "The zone you want to add the record in",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"keep_empty_zone": {
+				Type:        schema.TypeBool,
+				Description: "When destroy a ressource record, if a zone have only NS, delete the zone",
+				Optional:    true,
+				Default:     false,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The name of the record",
+				ForceNew:    true,
+				Required:    true,
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Description: "The type of the record",
+				ValidateFunc: validation.StringInSlice([]string{
+					domain.RecordTypeA.String(),
+					domain.RecordTypeAAAA.String(),
+					domain.RecordTypeALIAS.String(),
+					domain.RecordTypeCNAME.String(),
+					domain.RecordTypeMX.String(),
+					domain.RecordTypeNS.String(),
+					domain.RecordTypePTR.String(),
+					domain.RecordTypeSRV.String(),
+					domain.RecordTypeTXT.String(),
+					domain.RecordTypeTLSA.String(),
+					domain.RecordTypeCAA.String(),
+				}, false),
+				ForceNew: true,
+				Required: true,
+			},
+			"data": {
+				Type:        schema.TypeString,
+				Description: "The data of the record",
+				Required:    true,
+			},
+			"ttl": {
+				Type:         schema.TypeInt,
+				Description:  "The ttl of the record",
+				Optional:     true,
+				Default:      3600,
+				ValidateFunc: validation.IntAtLeast(60),
```

Comment:
please replace the validation for a `IntBetween`. The AtLeastCheck is getting timeouts  PATCHING with high numbers